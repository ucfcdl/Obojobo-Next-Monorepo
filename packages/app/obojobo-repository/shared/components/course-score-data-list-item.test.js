import React from 'react'
import { create } from 'react-test-renderer'

import CourseScoreDataListItem from './course-score-data-list-item'

describe('CourseScoreDataListItem', () => {
	const standardProps = {
		courseTitle: 'Mock Course Title',
		courseLabel: 'MCT-101',
		courseUserCount: 10,
		courseLastAccessed: 'Apr 1st 2020 - 9:15 AM',
		courseClick: jest.fn(),
		isSelected: false,
		index: 3
	}

	test('renders with standard expected props', () => {
		const component = create(<CourseScoreDataListItem {...standardProps} />)

		expect(component.toJSON()).toMatchSnapshot()
	})

	test('renders correctly when selected', () => {
		const component = create(<CourseScoreDataListItem {...standardProps} isSelected={true} />)

		expect(component.toJSON()).toMatchSnapshot()
	})

	test('user count displays properly without the "s" for 1 learner', () => {
		standardProps.courseUserCount = 1

		const component = create(<CourseScoreDataListItem {...standardProps} courseUserCount={1} />)

		expect(component.toJSON()).toMatchSnapshot()
	})

	test('onClick callback is called properly', () => {
		const component = create(<CourseScoreDataListItem {...standardProps} />)

		component.root.findByProps({ className: 'course-score-data-list--item' }).props.onClick()

		expect(standardProps.courseClick).toHaveBeenCalledTimes(1)
		expect(standardProps.courseClick).toHaveBeenCalledWith(standardProps.index)
	})
})
