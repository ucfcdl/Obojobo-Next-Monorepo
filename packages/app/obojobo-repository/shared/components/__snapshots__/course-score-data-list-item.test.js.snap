// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseScoreDataListItem renders correctly when selected 1`] = `
<div
  className="course-score-data-list--item is-selected"
  onClick={[Function]}
>
  <div
    className="courseTitle"
  >
    Mock Course Title
     (
    MCT-101
    )
  </div>
  <div
    className="userCount"
  >
    10
     Learner
    s
  </div>
  <div
    className="accessed"
  >
    Last Accessed   
    <span
      className="timestamp"
    >
      Apr 1st 2020 - 9:15 AM
    </span>
  </div>
</div>
`;

exports[`CourseScoreDataListItem renders with standard expected props 1`] = `
<div
  className="course-score-data-list--item"
  onClick={[Function]}
>
  <div
    className="courseTitle"
  >
    Mock Course Title
     (
    MCT-101
    )
  </div>
  <div
    className="userCount"
  >
    10
     Learner
    s
  </div>
  <div
    className="accessed"
  >
    Last Accessed   
    <span
      className="timestamp"
    >
      Apr 1st 2020 - 9:15 AM
    </span>
  </div>
</div>
`;

exports[`CourseScoreDataListItem user count displays properly without the "s" for 1 learner 1`] = `
<div
  className="course-score-data-list--item"
  onClick={[Function]}
>
  <div
    className="courseTitle"
  >
    Mock Course Title
     (
    MCT-101
    )
  </div>
  <div
    className="userCount"
  >
    1
     Learner
    
  </div>
  <div
    className="accessed"
  >
    Last Accessed   
    <span
      className="timestamp"
    >
      Apr 1st 2020 - 9:15 AM
    </span>
  </div>
</div>
`;
